### PR TITLE
Custom request weight extraction

### PR DESCRIPTION
**Pull request by joshgummersall**
This allows you to specify a function to extract the weight for a given
request.

### CODE REVIEW

File: src/express_middleware.coffee [hunk 1]
```diff
@@ -1,19 +1,28 @@
...
+    extractWeight or= @extractWeight
+
     (req, res, next) =>
-      @rateLimiter.incr extractIps(req), (err, isLimited) =>
-        if err
-          if @options.ignoreRedisErrors
-            isLimited = false
-          else
-            return next err
+      @rateLimiter.incr extractIps(req), extractWeight(req),
```
Comment (dudleycarr) [inline src/express_middleware.coffee]: We probably want to ensure that the weights are positive!

**Comment by dudleycarr** [comment]
ping

**Comment by joshgummersall** [comment]
ping

**Comment by dudleycarr** [comment]
lgtm

**Pull request merged**